ocaml/sdk-gen/common/CommonFunctions.ml

```diff
@@ -306,25 +306,111 @@ let json_releases =
     in
     try index_rec 0 list with Not_found -> -1
   in
-  let json_of_rel x =
+  let of_rel x =
     let y = version_index_of x unique_version_bumps + 1 in
-    `O
-      [
-        ( "code_name"
-        , `String (match x.code_name with Some r -> r | None -> "")
-        )
-      ; ("version_major", `Float (float_of_int x.version_major))
-      ; ("version_minor", `Float (float_of_int x.version_minor))
-      ; ("branding", `String x.branding)
-      ; ("version_index", `Float (float_of_int y))
-      ]
+    [
+      ("code_name", `String (Option.value x.code_name ~default:""))
+    ; ("version_major", `Float (float_of_int x.version_major))
+    ; ("version_minor", `Float (float_of_int x.version_minor))
+    ; ("branding", `String x.branding)
+    ; ("version_index", `Float (float_of_int y))
+    ]
+  in
+  let of_rels releases =
+    match releases with
+    | [] ->
+        `A []
+    | head :: tail ->
+        let head' = `O (("first", `Bool true) :: of_rel head) in
```

Comment:
What does `first` represent?

Reply:
A boolean for if it is the first parameter of a function. This is used to determine if a comma should be added before it.

Reply:
It's also equivalent to removing the trailing comma.

Reply:
I'm really confused. What do you mean by `function`? This field is added to all the release objects in an array, in which only the first and only one of them is true.

Regarding removal of trailing commas that part of the representation, not the content, I don't believe tit should be encoded in the json at all, unless I'm misunderstanding what you mean.

Reply:
For a Go function's parameters like:
```
foo(p1, p2, p3)
```
The template for the `(p1 type1, p2 type2, p3 type3)` looks like:
```
({{#func_params}}{{^first}}, {{/first}}{{name_internal}} {{type}}{{/func_params}})
```
The representation in OCaml is a list of params `[{p1; type1}; {p2; type2}; {p3; type3}]`. The template uses the `first` to determine that the first param doesn't need the comma before it after rendering:
```
{p1; type1} -> p1 type1
{p2; type2} -> , p2 type2
{p3; type3} -> , p3 type3
```

Reply:
I think a pattern that makes this obvious is to stringify the parameter list using ocaml, something like:

```ocaml
let go_func_params_of params =
  List.map ({name; typ; _} -> Printf.sprintf "%s %s" name typ) params |> List.concat ", "
in
...
`O [("typed_params", go_func_params_of params) :: params]
```
Then use it as is in mustache:
```mustache
({{#func_params}}{{typed_params}}{{name_internal}} {{type}}{{/func_params}})
```

Reply:
In Go SDK generation, the generating combined/complex source code (using `Printf.sprintf`) was avoided as much as possible. So it heavily relies on the template mechanisms. Unfortunately, the mustache doesn't support something like `List.concat`.

Reply:
I understand why you want to avoid using complex logic in ocaml and instead use mustache's templating features. Because this templating feature is missing, I believe using `String.concat` is justified in this case